docs/add_data_source.md

```diff
@@ -0,0 +1,151 @@
+
+### Data mapping <a name="mapping" />
+
+Review the [application profile](application_profile.md) to determine available and required fields.
+These are the field names that will be used in your configuration mapping.
+
+### Test data mapping <a name="test" />
+
+Add a `my_new_source_spec.rb` file to `spec/import`.
+
+#### Required test structure
+
+```
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe 'Transforming My New Data Source file' do
+  let(:indexer) { Pipeline.for('[MY_ID_USED_IN_SETTINGS]').indexer(HarvestedResource.new(original_filename: fixture_file_path)) }
+  let(:fixture_file_path) { File.join(fixture_path, 'csv/[my_new_fixture file]') }
+  let(:data) { File.open(fixture_file_path).read }
+  let(:exhibit) { create(:exhibit) }
+  let(:slug) { exhibit.slug }
+
+  before do
+    indexer.settings['exhibit_slug'] = slug
+    allow(CreateResourceJob).to receive(:perform_later)
+  end
+
+  it 'does the transform' do
+    indexer.process(data)
+    expect(CreateResourceJob).to have_received(:perform_later) do |_id, _two, json|
+      dlme = JSON.parse(json)
+      expect(dlme['agg_provider']).to eq 'EXPECTED VALUE FROM FIXTURE'
+      ..repeat for all fields to test..
+    end
+  end
+end
+```
+
+### Mapping configuration <a name="config" />
+
+Start by creating a new file in `lib/traject` named for the source: `my_new_source_config.rb`.
+
+#### Mapping CSV Data
+
+CSV configurations have the following requirements that must be included at the begining of
+your configuration file:
+
+```
+require_relative 'csv_reader'
+require_relative 'dlme_json_resource_writer'
+require_relative 'macros/dlme'
+require_relative 'macros/extraction'
+require_relative 'macros/csv'
+extend Macros::DLME
+extend Macros::Csv
+
+settings do
+  provide 'writer_class_name', 'DlmeJsonResourceWriter'
+  provide 'reader_class_name', 'CsvReader'
+end
+```
+
+Then, for each field in your mapping, a `to_field` command is required:
+
+`to_field('output field name'), column('csv_column_name', [OPTIONS])`
+
+#### Mapping XML Data
+
+XML configurations have the following requirements that must be included at the beginning of
+your configuration file:
+
+```
+equire_relative 'xml_reader'
+require_relative 'dlme_json_resource_writer'
+require_relative 'macros/dlme'
+require_relative 'macros/extraction'
+require_relative 'macros/xml'
+extend Macros::DLME
+extend Macros::Xml
+
+settings do
+  provide 'writer_class_name', 'DlmeJsonResourceWriter'
+  provide 'reader_class_name', 'XmlReader'
+end
+```
+Then, for each field in your mapping, a `to_field` command is required:
+
+`to_field('output field name'), extract_xml(xpath)`
+
+### Mapping macro <a name="macro" />
+
+This step is entirely optional and dependent on the needs of the particular data
+to be translated.
+
+Traject macros provide a facility for more complex data transformation. These are
+advanced settings, [review existing macros](../lib/traject/macros) for examples of
+macro structure.
+
```

Comment:
Maybe it's worth calling out the `transform_values` macro specifically, and how it is used for create the hierarchical data structures required for the `agg_is_shown_at` and `agg_is_shown_by` (any others?) keys?

Reply:
done.